span.go

```diff
@@ -52,6 +53,24 @@ type Span interface {
 	// See LogData for semantic details.
 	Log(data LogData)
 
+	// Return a PropagationInjector for the given `format` value, or nil if the
+	// Span does not support such a format.
+	//
+	// OpenTracing defines a common set of `format` values (see
+	// BuiltinPropagationFormat), and each has an expected carrier type.
+	//
+	// Other packages may declare their own `format` values, much like the keys
+	// used by the `net.Context` package (see
+	// https://godoc.org/golang.org/x/net/context#WithValue).
+	//
+	// Example usage (sans error handling):
+	//
+	//     span.PropagationInjectorForFormat(
+	//         opentracing.PROPAGATION_FORMAT_GO_HTTP_HEADER).InjectSpan(
+	//         span, httpReq.Header)
+	//
+	PropagationInjectorForFormat(format interface{}) PropagationInjector
```

Comment:
sorry, overlooked the `nil`. I guess that's ok, though it looks slightly non-idiomatic. Since the whole point of this exercise, maybe the `PropagationInjector` should have a chance to return an error, though: you don't know what it does, and while your standard impls will never hit a problem, others might.